doc/arch/adr-003-data-retention.md

```diff
@@ -0,0 +1,287 @@
+it will then need to be looked at as to what value storing them maintains, but
+hopefully by that point (expected to be a couple of years away) we'll have a
+much firmer understanding of the analytics usage and needs.
+
+SubscriptionLists don't currently have the ability to be deleted by apps, and
+it's likely if they gained that ability that would need to be some form of
+archiving so it could still be communicated to users that the list they had a
+subscription to is no longer available.
+
+#### Subscriber and Subscription
+
+Currently when a subscription is ended or we discover an email address is not
+valid we delete the subscriber and all subscriptions. This means we lose the
+ability to query to get a picture of data in the past; we lose information that
+can help debug account issues; and lose long term statistics on information on
+why/how people subscribe.
+
+The suggestion for tackling this is for us to persist the information related
+to Subscriptions and Subscribers indefinitely. As this is a somewhat low volume
+of data this shouldn't pose a huge issue storage wise.
+
+A field would be added to subscriber of `deactivated_at` which would be used to
+determine whether a subscriber is activated. Once a Subscriber has been
+deactivated for 28 days any personal information will be removed from the
+account. Any additional activity on the account will cause the account to be
+reactivated.
+
+_Aside: The `deactivated_at` field may not be required it could be determined
+by the lack of any active subscriptions. It just might be useful to have this
+as queries regarding end of a subscriber would be complex without it._
+
+This 28 day window is intended to provide us with sufficient time that we can
+deal with any support queries that occur due to account issues (as this is a
+reasonably regular occurance).
+
+With subscriptions the intention is to use this as an append only table and
+any change to a subscription (starting, ending, change of frequency) creates
+a new entry into this table. There will however be an active flag which can be
+used as part of the unique index on this to ensure a user can only have one
+active subscription at a time. This exception would not make it fully append
+anly.
+
+An additional change will be to add a timestamp, `ended_at`, which is used to
+record the time that the subscription ended at. Which will enable queries which
+determine the number of subscriptions at particular time periods. Additionally
+the suggestion is to include fields of the source of creating the subscription
+and to include the reason the subscription was ended.
+
+Finally we have a uuid field stored on subscription, it would likely be better
+for us to have a primary key that is the uuid and then not have a second
+numerical id.
+
+### Summary of database changes
+
+#### Email
+
+- Change the id type from a serial to a uuid
+- Add an archived_at timestamp
+- Would be removed from database after 14 days
```

Comment:
I suspect we're going to need to look into increasing disk space for 14 days, since we already had to clear out the table within about a week of running at 70%.

Reply:
Ah sure - do you reckon there is a time period that is acceptable where we don't have to up the disk space? As we could start at that and then increase that time if we learn from support queries we need to store them for longer.

If not, we probably should get a card in to up the disk space regardless.

Reply:
To be honest, I think we should get a card in about increasing disk space because at the moment I imagine we'll have to start deleting emails only after about 3 days.

Reply:
Cool I'll get a card in - I think we usually go up by about 64 GB that seem about reasonable?

Reply:
Seems reasonable to me!